immutable playbook

mbukatov opened the pull request:
This change default site.yml.example playbook file to be usable without changes. All ansible variables are now suggested to be placed in the inventory file (or variable file).

This is work in progress, waiting for feedback.

tendrl-bug-id: https://github.com/Tendrl/tendrl-ansible/issues/74

mbukatov committed:
drop system update from site playbook

mbukatov committed:
remove all defaults and docs from main playbook

mbukatov committed:
rename main playbook

mbukatov committed:
README unification, extension and cleanup

mbukatov commented:
@julienlim could you check if the new readme file makes sense to you?

https://github.com/mbukatov/tendrl-ansible/blob/031aa50450f3ee484e0218da54733292ee5f0559/README.md

mbukatov committed:
update ansible req. in spec file

cybernth commented on site.yml.sample:
```diff
@@ -13,18 +13,6 @@
-# Run yum update.
-#
-
-- hosts: all
```
Is there a specific reason why we are doing this?

mbukatov replied:
Previously, we expected that admin will edit the main playbook file so that he will notice what's going on there, commenting out or tweaking stuff to fit his needs. But this approach didn't work. So now, we no longer expect admin to look into the file and tweak it - which is what this pull request is about, and removal of "run yum update" task was done here as it's not obvious we are doing it in the playbook and some people may not want to run full system update during Tendrl installation.

cybernth commented on README.md:
I think its a good idea to talk about vaiables like:
tendrl_copr_repo
etcd auth related(etcd_tls_client_auth, etcd_cert_file etc)
The respective README files now, but atleast linking it to this file will be really helpful

mbukatov replied:
I could mention these variables in a section where I talk about optional variables. Right now it states:

> 4)  Add optional ansible variables into the invenory file.
>
>    Based on Tendrl documentation and description in README files of
>    tendrl-ansible roles, specify values for variables you like to tweak.
>
>    This is important because some features tendrl-ansible can help you
>    with are disabled by default as they require additional user input.
>
>    This includes etcd tls client authentication and tendrl notifier
>    configuration for snmp or smtp. Other features such as firewalld setup
>    for Tendrl can be disabled if needed.

Would your suggestion be addressed by mentioning the variables (` tendrl_copr_repo`, `etcd_tls_client_auth`, ...) in this section?

mbukatov replied:
@nthomas-redhat Does 752319f5aa31b036d0f9dee73615cca3e040beff address your suggestion?

mbukatov commented:
Based from @fbalak , I'm going to add a section related to expand cluster.
Update: added in commit 14228ae33e6b25382e758c6987dbb7aa859db20e

dahorak commented on README.md:
```diff
@@ -1,36 +1,70 @@
+* `tendrl-server`: installation of **Tendrl Server** machine (where Tendrl api,
+  web and etcd are running)
+* `tendrl-storage-node`: installation of **Tendrl Storage Node** machines
+  (Gluster servers, which you would like to monitor by Tendrl)
```
When installed form *rpm*, the roles contain prefix `tendrl-ansible.`, is it ok to reference them here in README file without the prefix? (If I didn't miss something, this README is not updated during package creation.)

julienlim replied:
For the entry about "tendrl-storage-node: installation of Tendrl Storage Node machines (Gluster servers, which you would like to monitor by Tendrl)..."

I might suggest some mention that the tendrl-storage node is the tendrl agent (aka node agent) as it's really an agent, and the "tendrl-storage-node" is may be confusing to the user.

mbukatov replied:
@dahorak yeah, good point - I need to unify the role names for both package and source repo
update: hopefully unified in 828f55bc52d46bfc26fb5ca6b4a0afeaf78622dc

dahorak commented on README.md:
```diff
@@ -39,55 +73,206 @@ tools (such as `ceph-installer` and `gdeploy`):
+as well.
+
+And last but not least, both `tendrl-server` and `tendrl-storage-node` contains
+many variables which one can use to tweak the installation. See README files of
```
To be clear, I would add ...both `tendrl-server` and `tendrl-storage-node` **roles** contains...

mbukatov replied:
Fixed in 818ac41985bdaab9a4be3c7847f94d57cbe5ed6d

dahorak commented on README.md:
```diff
@@ -39,55 +73,206 @@ tools (such as `ceph-installer` and `gdeploy`):
+
+## What installation steps from Tendrl installation documentation are not part of tendrl-ansible?
+
+This is should be clear from [Tendrl installation
```
Probably spare *is*... in the *This **is** should be clear...* sentence.

mbukatov replied:
fixed in a3091aa3f21b244c5d83fd5ec2dc8edcaf3e12c3

dahorak commented on README.md:
```diff
@@ -39,55 +73,206 @@ tools (such as `ceph-installer` and `gdeploy`):
+* Installation and configuration of GlusterFS on the machines, see
+  [gdeploy](https://gdeploy.readthedocs.io/en/latest/) for automation of this
+  task.
+* Setup of dedicated disk for etcd and graphite data directories.
```
If it would be possible, it would be nice to have link to the particular requirement here.

mbukatov replied:
I would rather do as few such links as possible. The guide is a moving target and linking into it may break soon. The whole guide is linked in the beginning of the section.

dahorak commented on README.md:
```diff
@@ -39,55 +73,206 @@ tools (such as `ceph-installer` and `gdeploy`):
+    * `etcd_fqdn` configure tendrl components to be able to connect to etcd
+      instance
+    * `graphite_fqdn` configures tendrl components (this value doesn't
+       reconfigure graphite itself!) to be able to connect to graphite instance
```
I would put the note in brackets at the end of the description:
```
* `graphite_fqdn` configures tendrl components to be able to connect to graphite instance
  (this value doesn't reconfigure graphite itself!)
```

mbukatov replied:
fixed in b4dfed13ba96ed5540cee907bfe7122611fe606e

dahorak commented on README.md:
```diff
@@ -39,55 +73,206 @@ tools (such as `ceph-installer` and `gdeploy`):
+    for Tendrl can be disabled if needed.
+
+5)  If you use tendrl-ansible from rpm package, copy `site.yml` playbook into
+    local directory (where you already store the inventory file):
```
I think, that *working* directory would be slightly better than *local* directory.

mbukatov replied:
ok, fixed in a2646c06cc54b48975468e76761fcd4aba23c454

dahorak commented on README.md:
The note about ssh keys is already mentioned in the step 6, so I'm not sure if it is required here?
And *cluster already installed* means Gluster cluster?

mbukatov replied:
Repetition in this context is not a bad thing, but good point about gluster, fix: c48bc68685e2ab7f355c07679f45333de13d6e88

dahorak commented on README.md:
I would use ``http://tendrl.example.com`` instead of ``http://ip.of.tendrl.server`` (maybe with some note describing what it is - pointing back to the example inventory file). I think it is not necessary to use IP address (instead of hostname).

mbukatov replied:
good point, fixed in: 1c9d195cef50c00c29b151a48d43de037fd8e8f8

dahorak reviewed (commented):
I would suggest just to fix few small typos and similar issues.

julienlim commented on README.md:
```diff
@@ -39,55 +73,206 @@ tools (such as `ceph-installer` and `gdeploy`):
+    * `etcd_fqdn` configure tendrl components to be able to connect to etcd
+      instance
+    * `graphite_fqdn` configures tendrl components (this value doesn't
+       reconfigure graphite itself!) to be able to connect to graphite instance
```
[all:vars]

This is awesome to now have this in the inventory file as opposed to editing them in the site.yml file!

julienlim commented on README.md:
```diff
@@ -39,55 +73,206 @@ tools (such as `ceph-installer` and `gdeploy`):
+    for Tendrl can be disabled if needed.
+
+5)  If you use tendrl-ansible from rpm package, copy `site.yml` playbook into
+    local directory (where you already store the inventory file):
```
"4)  Add optional ansible variables into the invenory file."

Can you provide specific examples for the optional ansible variables into the invenory (typo... should be inventory) file" i.e. tls client authentication and tendrl notified for SNMP or SMTP, firewalld?

mbukatov replied:
You mean something like this?

> This includes etcd tls client authentication (`etcd_tls_client_auth` and other variables), tendrl notifier configuration for snmp or smtp (`tendrl_notifier_email_id` and other variables), and other tweaks (eg. `tendrl_copr_repo` variable of `tendrl-copr` role).

mbukatov replied:
@julienlim Does 752319f5aa31b036d0f9dee73615cca3e040beff address your suggestion?

julienlim replied:
@mbukatov That works.  Thanks.

julienlim reviewed (commented):
@mbukatov I've reviewed and made some comments directly.

I've also another question/comment:

(1) Will you provide some guidance how someone can run the playbook as non-root?

I like the changes I read about, i.e. adding all:vars into the inventory file to make it easier to use the playbooks, support for production vs. non-production environments, instructions to run the precheck.yml, etc.

Nice work!

mbukatov committed:
readme: add examples of variables

Addressing feedback from Nishanth and Ju.

mbukatov committed:
readme: explain briefly cluster expand

Expand cluster section added based on suggestion from fbalak.

The procedure described in this commit is based on ongoing specification
for cluster expansion:

https://github.com/Tendrl/specifications/pull/254

That said, the expand procedure is not fully finished at the time of
writing this commit, see also:

https://github.com/Tendrl/commons/issues/849

mbukatov commented:
@fbalak @r0h4n I have added a brief description of expand into the readme file, see commit 14228ae33e6b25382e758c6987dbb7aa859db20e

mbukatov committed:
fix writing style

mbukatov committed:
fyx typo

mbukatov committed:
fix writing style

mbukatov committed:
fix writing style

mbukatov committed:
little clarification

mbukatov committed:
clarification

mbukatov committed:
fixing typos

fbalak commented on README.md:
I would add a mention about a notification that should be present after expand is finished: https://github.com/Tendrl/specifications/pull/254/files#diff-8e71bd954ebddb4a0ef4fd3c413d9f04R90

mbukatov replied:
I will add there more details, but when we have this resolved https://github.com/Tendrl/commons/issues/849

mbukatov committed:
add prefix to all tendrl-ansbile roles

This is needed to have the same role name for both rpm packaged
tendrl-ansible and the version straight from the git repository.

mbukatov committed:
explain how to use ansible as non-root remote user

mbukatov commented:
> (1) Will you provide some guidance how someone can run the playbook as non-root?

@julienlim I have added an example for this in 5f53a29588bc7564aa633f880a2bb3d5680bfff7

mbukatov committed:
mention grafana password lookup file

mbukatov committed:
polishing examples
